app/web/features/messages/groupchats/GroupChatView.tsx

```diff
@@ -197,101 +231,139 @@ export default function GroupChatView({ chatId }: { chatId: number }) {
+                          Chat members
+                        </MenuItem>,
+                        isChatAdmin
+                          ? [
+                              <MenuItem
+                                onClick={() => handleClick("admins")}
+                                key="admins"
+                              >
+                                Manage chat admins
+                              </MenuItem>,
+                              <MenuItem
+                                onClick={() => handleClick("settings")}
+                                key="settings"
+                              >
+                                Chat settings
+                              </MenuItem>,
+                            ]
+                          : null,
 
-                  {
-                    //<Menu> gives console warning for JSX Fragment children
-                    isChatAdmin && (
-                      <MenuItem onClick={() => handleClick("settings")}>
-                        Chat settings
-                      </MenuItem>
-                    )
-                  }
-                  {groupChat?.memberUserIdsList.includes(currentUserId) && (
-                    <MenuItem onClick={() => handleClick("leave")}>
-                      Leave chat
-                    </MenuItem>
-                  )}
-                </Menu>
-                {groupChat && (
-                  <>
-                    <InviteDialog
-                      open={isOpen.invite}
-                      onClose={() => handleClose("invite")}
-                      groupChat={groupChat}
-                    />
-                    <MembersDialog
-                      open={isOpen.members}
-                      onClose={() => handleClose("members")}
-                      groupChat={groupChat}
-                    />
-                    <AdminsDialog
-                      open={isOpen.admins}
-                      onClose={() => handleClose("admins")}
-                      groupChat={groupChat}
-                    />
-                    <GroupChatSettingsDialog
-                      open={isOpen.settings}
-                      onClose={() => handleClose("settings")}
-                      groupChat={groupChat}
-                    />
-                  </>
-                )}
-                <LeaveDialog
-                  open={isOpen.leave}
-                  onClose={() => handleClose("leave")}
-                  groupChatId={chatId}
-                />
-              </>
-            )}
+                        groupChat?.memberUserIdsList.includes(currentUserId) ? (
+                          <MenuItem
+                            onClick={() => handleClick("leave")}
+                            key="leave"
+                          >
+                            Leave chat
+                          </MenuItem>
+                        ) : null,
+                      ]
+                    : null,
+                ]}
```

Comment:
Is there any reason these are converted into list of JSX items as opposed to the `&&` conditional rendering as before? "It works" I suppose but a bit funky to read

Reply:
Yes, because `<Menu>` gives a warning if it has fragment children, and I didn't want to have 5 different `{ isDM && <MenuItem...`